Mathlib/Data/Seq/WSeq.lean

```diff
@@ -1363,8 +1362,7 @@ theorem destruct_ofSeq (s : Seq α) :
 
 @[simp]
 theorem head_ofSeq (s : Seq α) : head (ofSeq s) = Computation.pure s.head := by
-  simp only [head, Option.map_eq_map, destruct_ofSeq, Computation.map_pure, Option.map_map]
-  cases Seq.head s <;> rfl
+  simp [head]; cases Seq.head s <;> rfl
```

Comment:
@YaelDillies Why do you want to bring back a non-terminal `simp` here? If you don't like `simp only`, then we can use `simp? says simp only ...`

Reply:
Here is my comment for context: https://github.com/leanprover-community/mathlib4/pull/11259#discussion_r1527556127

My arguments for the squeezed simp being a worse proof are that
1. It mentions more lemmas
2. Those lemmas are part of the very basic API of their corresponding defs and could change if the defs change defeq
3. The non-terminal simp was just unfolding the definitions anyway (importantly, in a way that doesn't depend on the exact defeqs)